tools/bug-report/pkg/kubeclient/kubeclient.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeclient
+
+import (
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth" // Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// New creates a rest.Config qne Clientset from the given kubeconfig path and Context.
+func New(kubeconfig, kubeContext string) (*rest.Config, *kubernetes.Clientset, error) {
```

Comment:
Why don't we use pkg/kube instead of re-inventing?

Reply:
there's only 2% of that package that we actually need/use. imo it's cleaner just to copy a couple of functions rather than import that whole beast.

Reply:
The logic seems circular. You only need 2% of the package because you are not using it... Its not just about setting up the kube client, it already has everything you are trying to do here, from fetching info from Envoy to getting pod logs to fetching istiod debug info.